dispatch/dispatch.h

```diff
@@ -35,7 +35,7 @@
 #include <stdint.h>
 #include <stdbool.h>
 #include <stdarg.h>
-#if !defined(HAVE_UNISTD_H) || HAVE_UNISTD_H
+#if HAVE_UNISTD_H
```

Comment:
this doesn't work in public headers sadly because HAVE_UNISTD_H is a configure option, so doing that will break unix because HAVE_UNISTD_H is not defined.

there are two options, either use __has_include() but that requires clang support for the library user which I'm not a fan of (while requiring clang to build dispatch is a thing, requiring clang to build clients is another).

However, a thing we've started to do on the darwin side is to unifdef headers that we install.

I would be more than willing to accept a patch like that if the cmake build system is able to unifdef installed headers, and when installing on windows you'd `unifdef -DHAVE_UNISTD_H=0` and on unix `unifdef -DHAVE_UNISTD_H=1` and done.